### PR TITLE
DEVOPS-701 Introduce Java11 support

### PR DESCRIPTION
**Pull request by csongorhg**
This ticket serves as a basis for the Java 11 support, the changes include:
- upgrade Android Emulator 28 => 33
- upgrade Java 8 => Java 11
- upgrade SDK 28 => 33
- upgrade Kotlin 1.4.32 => 1.6.21
- upgrade Gradle 4.0.2 => 7.0.0
- upgrade Google services 4.3.2 => 4.3.10
- upgrade Arturbosch detekt 1.1.1 => 1.23.0
- upgrade Mocktio 2.8.47 => 3.12.4
- upgrade Work manager 2.5.0 => 2.7.1
- upgrade robolectric 4.2.1 => 4.7.3
- upgrade shadows-multidex 4.2.1 => 4.7.3
- upgrade Junit 1.1.1 => 1.1.3
- upgrade pmd 5.8.1 => 6.55.0
- upgrade checkstyle 8.26 => 10.12.1
- upgrade detekt 1.0.1 => 1.23.0
- use `catrobat/catrobat-android:api33` image in the Jenkins pipeline
- clean up gradle- and jenkinsfiles
- refactor `startEmulator `gradle task
- additional changes as a result of version changes

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [ ] Include the name of the Jira ticket in the PR’s title
- [ ] Include a summary of the changes plus the relevant context
- [ ] Choose the proper base branch (*java11-support*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no compiler or linter warnings
- [ ] Perform a self-review of the changes
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Confirm that new and existing unit tests pass locally
- [ ] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s gitflow workflow
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/AddBrickCatblocksTest.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2022 The Catrobat Team
```
Comment (mseiser) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/AddBrickCatblocksTest.java:3]: file moved on develop please exlude from commit

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/rtl/RTLOptionMenuIconsTest.kt [hunk 2]
```diff
@@ -131,8 +131,8 @@ class RTLOptionMenuIconsTest(
             val currentItem = optionsMenu.getItem(i)
             if (currentItem.isVisible) {
                 Assert.assertNotNull(currentItem.icon)
-                Assert.assertTrue(currentItem.icon.isVisible)
-                Assert.assertTrue(currentItem.icon.minimumWidth >= width)
+                currentItem.icon?.let { Assert.assertTrue(it.isVisible) }
```
Comment (mseiser) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/rtl/RTLOptionMenuIconsTest.kt:134]: Is this code change necessary?
also using !! should be only used as a last resort

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/AddBrickCatblocksTest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
```
Comment (mseiser) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/AddBrickCatblocksTest.java:7]: please remove code changes that aren't part of this ticket, this file was moved on develop, therefore my guess is that something went wrong when rebasing/merging

File: catroid/src/main/java/org/catrobat/catroid/stage/StageResourceHolder.java [hunk 4]
```diff
@@ -341,6 +343,7 @@ public boolean initFinished() {
 		return requiredResourceCounter == 0 && failedResources.isEmpty();
 	}
 
+	@SuppressLint("WrongConstant")
```
Comment (mseiser) [inline catroid/src/main/java/org/catrobat/catroid/stage/StageResourceHolder.java:346]: why do we need this supress?

**Review by mseiser — APPROVED** [review]
Some issues should be taken care of when we finally merging this PR into develop, for testing approved
Please don't merge into develop until remarks where resolved (not requested because this is a test)

**Comment by csongorhg** [comment]
@linearlinkoy can you please take a look at the proposed changes and fix them, thank you!

**Pull request merged**